java/k8s-addons/src/main/java/com/instaclustr/cassandra/k8s/SeedProvider.java

```diff
@@ -23,12 +26,40 @@ public SeedProvider(final Map<String, String> args) {
     @Override
     public List<InetAddress> getSeeds() {
         try {
-            final ImmutableList<InetAddress> seedAddresses = ImmutableList.copyOf(InetAddress.getAllByName(service));
+            // this gives us all IP addresses under cassandra-{metadata.name}-nodes
+            // each InetAddress will have a hostname set to "cassandra-{metadata.name}-nodes/ip.ad.dr.ess"
+            final ImmutableList<InetAddress> nodes = ImmutableList.copyOf(InetAddress.getAllByName(service));
 
-            logger.info("Discovered {} seed nodes: {}", seedAddresses.size(), seedAddresses);
 
-            return seedAddresses;
+            // We need to know what the hostname of each IP really is as now it
+            // seems as if there was "one" hostname (cassandra-{metadata.name}-nodes) and it had multiple IPs.
+            //
+            // We iterate over the list and resolve hostname of each such IP.
+            // This step will return host names like:
+            //
+            // cassandra-test-dc-cassandra-west1-a-0.cassandra-test-dc-cassandra-nodes.default.svc.cluster.local
+            //
+            // After that, seed nodes will be only the first nodes in a rack,
+            // host names start with the name of a respective pod and the first node
+            // in a rack always ends with "-0" suffix so we select only these.
+            final List<InetAddress> seeds = nodes.stream()
```

Comment:
This probably would just be easier to follow if it was just a regular for-loop rather than a stream pipeline.

Reply:
This would make dealing with exceptions easier too.